baseplate/lib/secrets.py

```diff
@@ -464,6 +467,88 @@ def _get_file_data(self, filename: str) -> Tuple[Dict, float]:
         return result
 
 
+class VaultCSIEntry(NamedTuple):
+    mtime: float
+    data: Any
+    # A mutex to prevent possible duplicate work from event loops
+    updating: bool
+
+
+class VaultCSISecretsStore(SecretsStore):
+    """Access to secret tokens with automatic refresh when changed.
+
+    This local vault allows access to the secrets cached on disk given a path to
+    a directory. It will automatically reload the cache when it is changed. Do not
+    cache or store the values returned by this class's methods but rather get
+    them from this class each time you need them. The secrets are served from
+    memory so there's little performance impact to doing so and you will be
+    sure to always have the current version in the face of key rotation etc.
+    """
+
+    path: Path
+    data_symlink: Path
+    cache: Dict[str, VaultCSIEntry]
+
+    def __init__(
+        self,
+        path: str,
+        parser: SecretParser,
+        timeout: Optional[int] = None,
+        backoff: Optional[float] = None,
+    ):  # pylint: disable=super-init-not-called
+        self.path = Path(path)
+        self.parser = parser
+        self.cache = {}
+        self.data_symlink = self.path.joinpath("..data")
+        if not self.path.is_dir():
+            raise ValueError(f"Expected {self.path} to be a directory.")
+        if not self.data_symlink.is_dir():
+            raise ValueError(
+                f"Expected {self.data_symlink} to be a directory. Verify {self.path} is the root of the Vault CSI mount."
+            )
+
+    def get_vault_url(self) -> str:
+        raise NotImplementedError
+
+    def get_vault_token(self) -> str:
+        raise NotImplementedError
+
```

Comment:
Please provide a comment regarding why there are unimplemented. I'm primarily concerned that somwhere someone is using one of these functions and adopting this upgrade will cause them to start to see errors. I would like clear notation in code and in docs indicating what's going on:)

Reply:
Worth noting that they are deprecated on the superclass, and the original vault implementation had them raising as well.